arangosh/Dump/DumpFeature.cpp

```diff
@@ -902,6 +860,85 @@ Result DumpFeature::runClusterDump(httpclient::SimpleHttpClient& client) {
 
   return {TRI_ERROR_NO_ERROR};
 }
+  
+Result DumpFeature::storeDumpJson(VPackSlice const& body,
+                                  std::string const& dbName) {
+  
+  // read the server's max tick value
+  std::string const tickString =
+  basics::VelocyPackHelper::getStringValue(body, "tick", "");
+  if (tickString == "") {
+    return ::ErrorMalformedJsonResponse;
+  }
+  LOG_TOPIC(INFO, Logger::DUMP)
+  << "Last tick provided by server is: " << tickString;
+  
+  // set the local max tick value
+  uint64_t maxTick = basics::StringUtils::uint64(tickString);
+  // check if the user specified a max tick value
+  if (_options.tickEnd > 0 && maxTick > _options.tickEnd) {
+    maxTick = _options.tickEnd;
+  }
+  
+  try {
+    VPackBuilder meta;
+    meta.openObject();
+    meta.add("database", VPackValue(dbName));
+    meta.add("lastTickAtDumpStart", VPackValue(tickString));
+    meta.close();
+    
+    // save last tick in file
+    auto file = _directory->writableFile("dump.json", true);
+    if (!::fileOk(file.get())) {
+      return ::fileError(file.get(), true);
+    }
+    
+    std::string const metaString = meta.slice().toJson();
+    file->write(metaString.c_str(), metaString.size());
+    if (file->status().fail()) {
+      return file->status();
+    }
+  } catch (basics::Exception const& ex) {
+    return {ex.code(), ex.what()};
+  } catch (std::exception const& ex) {
+    return {TRI_ERROR_INTERNAL, ex.what()};
+  } catch (...) {
+    return {TRI_ERROR_OUT_OF_MEMORY, "out of memory"};
```

Comment:
should it be `TRI_ERROR_INTERNAL` rather than OOM?

Reply:
I just moved this code, I don't know why we return OOM here